ts/main.ts

```diff
@@ -0,0 +1,226 @@
+const $welcomePage = document.querySelector('#welcome-page') as HTMLDivElement;
+const $searchResults = document.querySelector(
+  '#search-results',
+) as HTMLDivElement;
+const $marketmonLogo = document.querySelector(
+  '#marketmon-logo',
+) as HTMLImageElement;
+const $searchInput = document.querySelector('#search-input') as HTMLFormElement;
+const $cardContainerRow = document.querySelector(
+  '#card-container-row',
+) as HTMLDivElement;
+
+if (!$form) throw new Error('the $from query failed.');
+if (!$welcomePage) throw new Error('the $welcomePage query failed');
+if (!$searchResults) throw new Error('the $searchResults query failed');
+if (!$marketmonLogo) throw new Error('the $marketmonLogo query failed');
+if (!$searchInput) throw new Error('the $searchInput query failed.');
+if (!$cardContainerRow) throw new Error('the $cardContainerRow query failed.');
+
+// EVENT LISTENER: to listen for when the MarketMon logo is clicked
+
+$marketmonLogo.addEventListener('click', () => {
+  viewSwap('welcome-page');
+});
+
+// EVENT LISTENER: to listen for when a search is submitted
+
+$form.addEventListener('submit', (event: Event): void => {
+  event.preventDefault();
+  viewSwap('search-results');
+  $cardContainerRow.innerHTML = '';
+  const nameOfPokemonSearched = $searchInput.value;
+  if (nameOfPokemonSearched === '') {
+    displayNoMatches();
+  } else {
+    fetchCards(nameOfPokemonSearched);
+  }
+});
+
+// FUNCTION: to swap views
+
+function viewSwap(view: string): void {
+  const valueOfView = view;
+  data.view = valueOfView;
+
+  if (view === 'welcome-page') {
+    $welcomePage.className = 'row container show';
+    $searchResults.className = 'hidden';
+  } else if (view === 'search-results') {
+    $welcomePage.className = 'hidden';
+    $searchResults.className = 'show';
+  }
+}
+
+// FUNCTION: to fetch information from api
+
+async function fetchCards(searchCriteria: unknown): Promise<void> {
+  try {
+    const response = await fetch(
+      `https://api.pokemontcg.io/v2/cards?q=name:${searchCriteria}`,
+    );
+    if (!response.ok) {
+      throw new Error('Network response was not ok');
+    }
+    const cardObjects = await response.json();
+    if (searchCriteria === '') {
+      displayNoMatches();
+    } else {
+      renderCards(cardObjects);
+    }
+  } catch (error) {
+    console.log('Fetch function failed: ', error);
+  }
+}
+
+// FUNCTION: to render cards for search result
+
+function renderCards(cardObjects: any): any {
+  // created an array of objects that each store the information of a single card
+
+  const cardObjectsData = cardObjects.data;
+  const cardInfoArray: any = [];
+
+  for (let i = 0; i < cardObjectsData.length; i++) {
+    const cardInfo: any = {
+      smallImage: cardObjects.data[i].images.small,
+      cardName: cardObjects.data[i].name,
+      setName: cardObjects.data[i].set.name,
+      cardNumber: cardObjects.data[i].number,
+    };
```

Comment:
Rather than using `any` for this, it would be better to create an `interface` for this object